ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -212,9 +212,9 @@ data PeerSelectionActions peeraddr peerconn m = PeerSelectionActions {
 -- | Callbacks which are performed to change peer state.
 --
 data PeerStateActions peeraddr peerconn m = PeerStateActions {
-    -- | Monitor peer state.
+    -- | Monitor peer state.  Must be non-blocking.
     --
-    monitorPeerConnection    :: peerconn -> STM m (PeerStatus, ReconnectDelay),
+    monitorPeerConnection    :: peerconn -> STM m (PeerStatus, Maybe ReconnectDelay),
```

Comment:
If it must be non-blocking could we somehow get rid of `STM`?

Reply:
no because the `Guarded` monoid is executed in `STM` monad.

Reply:
Right, I had edited this comment don't know why it didn't get through. But I think this comment is misleading, we want `monitorPeerConnection` to be non-blocking in a particular function's logic. Saying this at the top level and then have a `STM` type is bit confusing. I'd be more accurate and explain in what situation we do not want to block